crates/test-utils/src/hive_utils/madara/utils.rs

```diff
@@ -257,37 +266,35 @@ mod tests {
-            // Get lock on the Starknet sequencer
-            let mut starknet = env.sequencer().sequencer.backend.state.blocking_write();
-            let mut counter_storage = HashMap::new();
-
-            // Set the counter bytecode length into the contract
-            let key = get_starknet_storage_key("bytecode_len_", &[]);
-            let value = Into::<StarkFelt>::into(StarkFelt::from(deployed_evm_bytecode_len as u64));
+
+        // Get lock on the Starknet sequencer
+        let mut starknet = env.sequencer().sequencer.backend.state.write().await;
+        let mut counter_storage = HashMap::new();
+
+        // Set the counter bytecode length into the contract
+        let key = get_starknet_storage_key("bytecode_len_", &[]);
+        let value = Into::<StarkFelt>::into(StarkFelt::from(deployed_evm_bytecode_len as u64));
+        counter_storage.insert(key, value);
+
+        // Set the counter bytecode into the contract
+        counter_genesis_storage.into_iter().for_each(|((_, k), v)| {
+            let key = StarknetStorageKey(Into::<StarkFelt>::into(k.0).try_into().unwrap());
+            let value = Into::<StarkFelt>::into(v.0);
             counter_storage.insert(key, value);
+        });
 
-            // Set the counter bytecode into the contract
-            counter_genesis_storage.into_iter().for_each(|((_, k), v)| {
-                let key = StarknetStorageKey(Into::<StarkFelt>::into(k.0).try_into().unwrap());
-                let value = Into::<StarkFelt>::into(v.0);
-                counter_storage.insert(key, value);
-            });
-
-            // Deploy the contract account at genesis address
-            let contract_account_class_hash = env.kakarot().contract_account_class_hash;
-            let counter_address =
-                StarknetContractAddress(Into::<StarkFelt>::into(counter_genesis_address).try_into().unwrap());
-
-            starknet.set_class_hash_at(counter_address, ClassHash(contract_account_class_hash.into())).unwrap();
-            starknet.set_nonce(counter_address, Nonce(StarkFelt::from(1u8)));
-            for (key, value) in counter_storage.into_iter() {
-                starknet.set_storage_at(counter_address, key, value);
-            }
-        })
-        .await
-        .unwrap();
+        // Deploy the contract account at genesis address
+        let contract_account_class_hash = env.kakarot().contract_account_class_hash;
+        let counter_address =
+            StarknetContractAddress(Into::<StarkFelt>::into(counter_genesis_address).try_into().unwrap());
+
+        starknet.set_class_hash_at(counter_address, ClassHash(contract_account_class_hash.into())).unwrap();
+        starknet.set_nonce(counter_address, Nonce(StarkFelt::from(1u8)));
+        for (key, value) in counter_storage.into_iter() {
+            starknet.set_storage_at(counter_address, key, value);
+        }
+        // Need to drop the lock on the sequencer to avoid deadlock, so we can then get the bytecode
```

Comment:
We should have discussion somewhere about all these async issues we encounter and how we solve them.